spec/unit/indexer_spec.rb

```diff
@@ -0,0 +1,61 @@
+require "spec_helper"
+
+RSpec.describe Indexer do
+  describe ".find_content_id" do
+    let(:logger) { Logging.logger[described_class] }
+    let(:content_id) { SecureRandom.uuid }
+    let(:base_path) { "/base-path" }
+
+    context "when publishing api has the content item" do
+      it "returns the content id" do
+        stub_publishing_api_has_lookups(base_path => content_id)
+        expect(described_class.find_content_id(base_path, logger)).to eq(content_id)
+      end
+    end
+
+    context "when the lookup times out" do
+      before do
+        publishing_api_adapter = instance_double(GdsApi::PublishingApi)
+        allow(GdsApi::PublishingApi)
+          .to receive(:new)
+          .and_return(publishing_api_adapter)
+        allow(publishing_api_adapter)
+          .to receive(:lookup_content_id)
+          .and_raise(GdsApi::TimedOutException)
```

Comment:
It's a shame there's not a stub [here](https://github.com/alphagov/gds-api-adapters/blob/721512aace1930b7475fe563a421c3dee6fb7c9d/lib/gds_api/test_helpers/publishing_api.rb#L181-L189)! Can keep it in mind to add one if we end up writing this code another time.